test/e2e_test.go

```diff
@@ -97,6 +101,9 @@ var _ = Describe("MetricsServer", func() {
 		if err != nil {
 			panic(err)
 		}
+		// When the resource consumer is just started, the cpu occupancy rate is not accurate.
+		// So we need to wait for metrics-server for 3 scrape cycles to ensure the accuracy of metrics
+		time.Sleep(45 * time.Second)
```

Comment:
Metrics server should provide an accurate metrics from it's start. If not it means the feature for checking contrainer start time is broken.

Reply:
Yeah, Metrics server  provides an accurate metrics, this is correct..
I mean the cpu actually consumed by the cpu-consumer pod is inaccurate at the beginning
For example, we set
`"--duration-sec=60", "--millicores=50"`
In fact, the cpu consumption is as follows

time elapse: 118.769µs cpu real consumption: 1436.7816091952816
time elapse: 11.471613ms cpu real consumption: 29.770100399663576
time elapse: 16.732648ms cpu real consumption: 350.02822128859964
time elapse: 27.074909ms cpu real consumption: 226.36449423762713
time elapse: 37.45827ms cpu real consumption: 172.6597078175341
time elapse: 47.780835ms cpu real consumption: 141.80841070805468
time elapse: 58.599994ms cpu real consumption: 120.37636512780927
time elapse: 70.457176ms cpu real consumption: 104.0473953373914
time elapse: 81.324634ms cpu real consumption: 93.25414081646278
time elapse: 92.369565ms cpu real consumption: 85.08307521439265
time elapse: 103.450842ms cpu real consumption: 78.36805062818036
time elapse: 113.863051ms cpu real consumption: 73.53060081649919
time elapse: 124.224052ms cpu real consumption: 69.96611341753581
time elapse: 134.611746ms cpu real consumption: 67.12556554404631
time elapse: 144.915772ms cpu real consumption: 64.38021815929622
time elapse: 155.227015ms cpu real consumption: 61.87126119115052
time elapse: 165.563837ms cpu real consumption: 59.32510053527889
time elapse: 175.864924ms cpu real consumption: 57.273190512833104
time elapse: 186.237777ms cpu real consumption: 55.20465861537461
time elapse: 196.625293ms cpu real consumption: 54.34333604727722
time elapse: 206.966914ms cpu real consumption: 53.174968906614986
time elapse: 218.457077ms cpu real consumption: 51.58767883946246
time elapse: 229.46106ms cpu real consumption: 50.33885607068835
time elapse: 240.224412ms cpu real consumption: 49.3819729050328
time elapse: 244.932831ms cpu real consumption: 67.7711642755581
time elapse: 255.275188ms cpu real consumption: 65.81251602761705
time elapse: 268.887462ms cpu real consumption: 64.0828375768856
time elapse: 279.2125ms cpu real consumption: 62.69802466503529
time elapse: 290.116807ms cpu real consumption: 61.39990571093338
time elapse: 301.000128ms cpu real consumption: 60.117124152740686
time elapse: 311.339563ms cpu real consumption: 58.98357231503753
time elapse: 321.684322ms cpu real consumption: 58.12611590185177
time elapse: 334.568441ms cpu real consumption: 56.72019052815947
time elapse: 348.115012ms cpu real consumption: 55.445638680921434
time elapse: 363.486174ms cpu real consumption: 53.845929185554844
time elapse: 376.442045ms cpu real consumption: 52.60396468069626
time elapse: 387.639495ms cpu real consumption: 51.81682123016297
time elapse: 398.025608ms cpu real consumption: 50.997078845804225
time elapse: 408.446094ms cpu real consumption: 50.2708251926321
time elapse: 420.444045ms cpu real consumption: 49.49743496507179
time elapse: 425.645363ms cpu real consumption: 60.82189380227534
time elapse: 437.284286ms cpu real consumption: 59.87438576682197
time elapse: 450.478193ms cpu real consumption: 58.643877768119935
time elapse: 464.434409ms cpu real consumption: 57.32864458334455
time elapse: 475.579667ms cpu real consumption: 56.536116750991525
time elapse: 486.062878ms cpu real consumption: 56.010444586500384
time elapse: 498.343219ms cpu real consumption: 55.258689026402365
time elapse: 510.381175ms cpu real consumption: 54.44224142192646
time elapse: 524.849892ms cpu real consumption: 53.43519471086829
time elapse: 535.565498ms cpu real consumption: 52.879383372264755
time elapse: 548.193372ms cpu real consumption: 52.11018820448377
time elapse: 562.469147ms cpu real consumption: 51.13895957048779
time elapse: 574.46256ms cpu real consumption: 50.486486442626735
time elapse: 584.811803ms cpu real consumption: 50.149350725136834
time elapse: 597.605885ms cpu real consumption: 49.478402288262124
time elapse: 602.360234ms cpu real consumption: 56.838196119236024
time elapse: 612.608214ms cpu real consumption: 56.195829838067226
time elapse: 624.556998ms cpu real consumption: 55.51188941261642
time elapse: 635.775117ms cpu real consumption: 54.91829927103899
time elapse: 648.177416ms cpu real consumption: 54.33935689623421
time elapse: 663.402575ms cpu real consumption: 53.54594888010854
time elapse: 673.868805ms cpu real consumption: 53.06157321478858
time elapse: 686.473712ms cpu real consumption: 52.513333454465645
time elapse: 697.080589ms cpu real consumption: 52.20581783820712
time elapse: 709.762632ms cpu real consumption: 51.69294303897235
time elapse: 720.327381ms cpu real consumption: 51.40163545290853
time elapse: 731.735117ms cpu real consumption: 50.949404578554564
time elapse: 748.075171ms cpu real consumption: 50.27365347719588
time elapse: 769.464824ms cpu real consumption: 49.30590339807559
time elapse: 775.023881ms cpu real consumption: 56.00092177179143
time elapse: 786.926206ms cpu real consumption: 55.42061139067801
time elapse: 797.268206ms cpu real consumption: 54.99743874268228
time elapse: 808.941317ms cpu real consumption: 54.59545451957901
time elapse: 819.449968ms cpu real consumption: 54.35833341470322
time elapse: 834.483321ms cpu real consumption: 53.75462641859029
time elapse: 846.471256ms cpu real consumption: 53.326942529000696
time elapse: 859.47211ms cpu real consumption: 52.910612305809074
time elapse: 871.457831ms cpu real consumption: 52.51469785750189
time elapse: 883.445817ms cpu real consumption: 52.127960219965544
time elapse: 894.623759ms cpu real consumption: 51.77893413558807
time elapse: 906.557958ms cpu real consumption: 51.50540402887596
time elapse: 918.125883ms cpu real consumption: 51.20181503075855
time elapse: 929.449094ms cpu real consumption: 50.850203818702454
time elapse: 941.482312ms cpu real consumption: 50.47878996304419
time elapse: 952.418699ms cpu real consumption: 50.18789798538166
time elapse: 965.528985ms cpu real consumption: 49.78705030099822
time elapse: 970.059358ms cpu real consumption: 54.184953027308666
time elapse: 983.502045ms cpu real consumption: 53.65350732149607
time elapse: 994.498902ms cpu real consumption: 53.27363784241284
time elapse: 1.007416056s cpu real consumption: 52.877993922799185
time elapse: 1.01847595s cpu real consumption: 52.52129020207918
time elapse: 1.030388925s cpu real consumption: 52.162553810614185
time elapse: 1.041395601s cpu real consumption: 51.83908351381418
time elapse: 1.053414628s cpu real consumption: 51.50947845894479
time elapse: 1.066411264s cpu real consumption: 51.22350664135333
time elapse: 1.076741194s cpu real consumption: 51.01982868565287
time elapse: 1.088511495s cpu real consumption: 50.69617123003832
time elapse: 1.102395287s cpu real consumption: 50.38880564376833
time elapse: 1.114129742s cpu real consumption: 50.119215860235556
time elapse: 1.125298242s cpu real consumption: 49.87120777478975
time elapse: 1.130373223s cpu real consumption: 53.86717083150945
time elapse: 1.142559947s cpu real consumption: 53.58402277708076
time elapse: 1.153234658s cpu real consumption: 53.41982317545067
time elapse: 1.165119871s cpu real consumption: 53.14127186414047
time elapse: 1.176542906s cpu real consumption: 52.92229400095674
time elapse: 1.187618055s cpu real consumption: 52.784946370009365
time elapse: 1.198346921s cpu real consumption: 52.66306843845796
time elapse: 1.208742022s cpu real consumption: 52.49454134093241
time elapse: 1.220922722s cpu real consumption: 52.24559737353326
time elapse: 1.233664075s cpu real consumption: 51.99779726323805
time elapse: 1.246912591s cpu real consumption: 51.834206265614725
time elapse: 1.259783972s cpu real consumption: 51.533856784092734
time elapse: 1.271508687s cpu real consumption: 51.294454127121824
time elapse: 1.283389023s cpu real consumption: 51.06171904629754
time elapse: 1.296562709s cpu real consumption: 50.84283121575817
time elapse: 1.306905114s cpu real consumption: 50.783903496358995
time elapse: 1.317145571s cpu real consumption: 50.54056134485706
time elapse: 1.327949323s cpu real consumption: 50.28778954186306
time elapse: 1.34041019s cpu real consumption: 49.98794840389277
time elapse: 1.345386727s cpu real consumption: 52.98546569673296
time elapse: 1.358076607s cpu real consumption: 52.62737890397954
time elapse: 1.370175775s cpu real consumption: 52.33213284844871
time elapse: 1.380484918s cpu real consumption: 52.09775082303224
time elapse: 1.390883963s cpu real consumption: 51.908171350768335
time elapse: 1.401519081s cpu real consumption: 51.73377864443304
time elapse: 1.411927658s cpu real consumption: 51.51972682803813
time elapse: 1.423095843s cpu real consumption: 51.27933994459789
time elapse: 1.433468208s cpu real consumption: 51.09287279555118
time elapse: 1.445648467s cpu real consumption: 50.93403268431723
time elapse: 1.456615648s cpu real consumption: 50.739703649824996
time elapse: 1.469076914s cpu real consumption: 50.47372797325088
time elapse: 1.485437184s cpu real consumption: 50.0843382702643
time elapse: 1.497522335s cpu real consumption: 49.83412787701799
time elapse: 1.50195322s cpu real consumption: 52.63093158047943
time elapse: 1.512365448s cpu real consumption: 52.41658406599778
time elapse: 1.522840582s cpu real consumption: 52.238239557824706
time elapse: 1.535464245s cpu real consumption: 52.07041299627485
time elapse: 1.547794609s cpu real consumption: 51.84945115225311
time elapse: 1.55948785s cpu real consumption: 51.621551633808984
time elapse: 1.57009668s cpu real consumption: 51.46143716940513
time elapse: 1.581793805s cpu real consumption: 51.285600692044156
time elapse: 1.592030389s cpu real consumption: 51.136897622903234
time elapse: 1.602771318s cpu real consumption: 50.98827524061748
time elapse: 1.614597529s cpu real consumption: 50.77199119527488
time elapse: 1.625262325s cpu real consumption: 50.63276080739498
time elapse: 1.63882924s cpu real consumption: 50.443184973020294
time elapse: 1.651415696s cpu real consumption: 50.31388002449137
time elapse: 1.664439104s cpu real consumption: 50.085730249189965
time elapse: 1.675720309s cpu real consumption: 49.95254409835546
time elapse: 1.680939054s cpu real consumption: 52.794953190851636
time elapse: 1.691382273s cpu real consumption: 52.57935644248939
time elapse: 1.704402316s cpu real consumption: 52.3922964103226
time elapse: 1.715104162s cpu real consumption: 52.2243070208397
time elapse: 1.725404131s cpu real consumption: 52.09209659873119
time elapse: 1.735802058s cpu real consumption: 51.960978258285905
time elapse: 1.747010577s cpu real consumption: 51.79518469372883
time elapse: 1.757378388s cpu real consumption: 51.63587741137373
time elapse: 1.767725737s cpu real consumption: 51.47267467415904
time elapse: 1.778703271s cpu real consumption: 51.31607160967444
time elapse: 1.789385745s cpu real consumption: 51.157661060334675
time elapse: 1.801432513s cpu real consumption: 50.995349421991804
time elapse: 1.813203345s cpu real consumption: 50.82590484852189
time elapse: 1.823566935s cpu real consumption: 50.67082164223257
time elapse: 1.83811068s cpu real consumption: 50.455624935680746
time elapse: 1.848689191s cpu real consumption: 50.38774494215501
time elapse: 1.859627602s cpu real consumption: 50.34607526007882
time elapse: 1.870010962s cpu real consumption: 50.21082317849406
time elapse: 1.881533536s cpu real consumption: 50.09591738748158
time elapse: 1.893334203s cpu real consumption: 49.9772099907996
time elapse: 1.898785397s cpu real consumption: 52.51907414700709
time elapse: 1.910591929s cpu real consumption: 52.38874283667183
time elapse: 1.922943473s cpu real consumption: 52.25593057653923
time elapse: 1.93339295s cpu real consumption: 52.13874841109798
time elapse: 1.944512967s cpu real consumption: 52.00566645652694
time elapse: 1.957533494s cpu real consumption: 51.803105246996026
time elapse: 1.969470142s cpu real consumption: 51.6261028345669
time elapse: 1.981980043s cpu real consumption: 51.412335979090855
time elapse: 1.992451003s cpu real consumption: 51.26759464037899
time elapse: 2.004519215s cpu real consumption: 51.08228820812597
time elapse: 2.014762534s cpu real consumption: 50.91510194295064
time elapse: 2.027772775s cpu real consumption: 50.69986158171858
time elapse: 2.038501453s cpu real consumption: 50.55585579639612
time elapse: 2.052396924s cpu real consumption: 50.49305870543998
time elapse: 2.062742543s cpu real consumption: 50.3560783677532
time elapse: 2.073108576s cpu real consumption: 50.247609250570875
time elapse: 2.083499632s cpu real consumption: 50.14772605902734
time elapse: 2.094870856s cpu real consumption: 50.04789373282162
time elapse: 2.105646427s cpu real consumption: 49.93256062282961
time elapse: 2.110343856s cpu real consumption: 51.94297692884416
time elapse: 2.122430153s cpu real consumption: 51.80137339398374
time elapse: 2.132812526s cpu real consumption: 51.691157783677966
time elapse: 2.143313358s cpu real consumption: 51.55241966767122
time elapse: 2.155948901s cpu real consumption: 51.36903673518185
time elapse: 2.166589198s cpu real consumption: 51.24938200849457
time elapse: 2.177429259s cpu real consumption: 51.1213471117322
time elapse: 2.18871581s cpu real consumption: 51.000146296614744
time elapse: 2.199410209s cpu real consumption: 50.839653380589525
time elapse: 2.209748433s cpu real consumption: 50.70097793840597
time elapse: 2.220068894s cpu real consumption: 50.5709551833231
time elapse: 2.232513239s cpu real consumption: 50.38105464531885
time elapse: 2.245501409s cpu real consumption: 50.197989830936805
time elapse: 2.258474426s cpu real consumption: 50.05815774275195
time elapse: 2.270480768s cpu real consumption: 49.93319683528458
time elapse: 2.274891525s cpu real consumption: 51.78336363618732
time elapse: 2.286974897s cpu real consumption: 51.61734724944921
time elapse: 2.298474843s cpu real consumption: 51.45393301822295
time elapse: 2.310449589s cpu real consumption: 51.28789455683963
time elapse: 2.323451086s cpu real consumption: 51.13289051384146
time elapse: 2.335449513s cpu real consumption: 50.97664670563425
time elapse: 2.350616535s cpu real consumption: 50.771892590235375
time elapse: 2.362178538s cpu real consumption: 50.670381423807775
time elapse: 2.376865508s cpu real consumption: 50.46776212021082
time elapse: 2.387797484s cpu real consumption: 50.34214734856079
time elapse: 2.401697498s cpu real consumption: 50.173090416394714
time elapse: 2.412596828s cpu real consumption: 50.0272394314469
time elapse: 2.424423492s cpu real consumption: 49.904919545493385
time elapse: 2.428998776s cpu real consumption: 51.76993954978404
time elapse: 2.439466782s cpu real consumption: 51.62244762341763
time elapse: 2.449808413s cpu real consumption: 51.487452298213576
time elapse: 2.460126745s cpu real consumption: 51.37142380465698
time elapse: 2.470432952s cpu real consumption: 51.252805048079026
time elapse: 2.481081026s cpu real consumption: 51.16292365755656
time elapse: 2.491421981s cpu real consumption: 51.04893333440849
time elapse: 2.502499502s cpu real consumption: 50.929145990133335
time elapse: 2.513398612s cpu real consumption: 50.79329386364393
time elapse: 2.525441083s cpu real consumption: 50.67273332057785
time elapse: 2.536238775s cpu real consumption: 50.56000118278526
time elapse: 2.548157564s cpu real consumption: 50.47635992772933
time elapse: 2.558482421s cpu real consumption: 50.39166991893325
time elapse: 2.56884699s cpu real consumption: 50.326166436918236
time elapse: 2.579189883s cpu real consumption: 50.24170781689827
time elapse: 2.589542147s cpu real consumption: 50.12680452204733
time elapse: 2.599897409s cpu real consumption: 50.051796414412905
time elapse: 2.610163148s cpu real consumption: 49.97810560896579
time elapse: 2.614700718s cpu real consumption: 51.54933846458803
time elapse: 2.624983764s cpu real consumption: 51.418441386773
time elapse: 2.635363273s cpu real consumption: 51.31801286154789
time elapse: 2.645692734s cpu real consumption: 51.23259260025763
time elapse: 2.656162189s cpu real consumption: 51.11692570350459
time elapse: 2.666807255s cpu real consumption: 51.00791482186729
time elapse: 2.679273369s cpu real consumption: 50.894248350677714
time elapse: 2.695194001s cpu real consumption: 50.70387618794305
time elapse: 2.705518195s cpu real consumption: 50.59923729996975
time elapse: 2.715857598s cpu real consumption: 50.50334235659176
time elapse: 2.7262015s cpu real consumption: 50.40798968716439
time elapse: 2.742409135s cpu real consumption: 50.29117801857308
time elapse: 2.753421268s cpu real consumption: 50.18757441133464
time elapse: 2.764475414s cpu real consumption: 50.08380894119768
time elapse: 2.778705672s cpu real consumption: 50.006476545368514
time elapse: 2.794430565s cpu real consumption: 49.85197409621398
time elapse: 2.799234578s cpu real consumption: 51.46281355956539
time elapse: 2.811424626s cpu real consumption: 51.308790310989245
time elapse: 2.821612768s cpu real consumption: 51.198676012502446
time elapse: 2.835428656s cpu real consumption: 51.15301996605691
time elapse: 2.846423342s cpu real consumption: 51.06327350698337
time elapse: 2.859532335s cpu real consumption: 50.955196374271885
time elapse: 2.869870355s cpu real consumption: 50.87416506318253
time elapse: 2.880527547s cpu real consumption: 50.7585798324908
time elapse: 2.894511186s cpu real consumption: 50.61558963843907
time elapse: 2.905478368s cpu real consumption: 50.56016273138702
time elapse: 2.916518514s cpu real consumption: 50.49275186086394
time elapse: 2.931497057s cpu real consumption: 50.392680886642076
time elapse: 2.941939311s cpu real consumption: 50.31200669409465
time elapse: 2.954058895s cpu real consumption: 50.26299550769535
time elapse: 2.965929683s cpu real consumption: 50.16810660256133
time elapse: 2.976273777s cpu real consumption: 50.08535295317674
time elapse: 2.987219769s cpu real consumption: 49.99985124412373
time elapse: 2.992120336s cpu real consumption: 51.516608737915696
time elapse: 3.002480407s cpu real consumption: 51.410249669913455
time elapse: 3.013017643s cpu real consumption: 51.31485772206669
time elapse: 3.023360516s cpu real consumption: 51.22081304876139
time elapse: 3.033673305s cpu real consumption: 51.11940006166864
time elapse: 3.043978612s cpu real consumption: 51.0200984228877
time elapse: 3.054287186s cpu real consumption: 50.91984674597241
time elapse: 3.064508927s cpu real consumption: 50.82455175496783
time elapse: 3.075327064s cpu real consumption: 50.73329390780849
time elapse: 3.087437365s cpu real consumption: 50.630903643475165
time elapse: 3.099432705s cpu real consumption: 50.516783513991356
time elapse: 3.109857201s cpu real consumption: 50.4383641774831
time elapse: 3.120472091s cpu real consumption: 50.339767697717065
time elapse: 3.132742418s cpu real consumption: 50.22885051558744
time elapse: 3.143112369s cpu real consumption: 50.16032655127914
time elapse: 3.159448958s cpu real consumption: 49.97896327195673
time elapse: 3.164663813s cpu real consumption: 51.527693897869675
time elapse: 3.175211705s cpu real consumption: 51.426554460629525
time elapse: 3.186325457s cpu real consumption: 51.3540361502031

Reply:
This sleep sounds actually fair to me based on the data above.

> Metrics server should provide an accurate metrics from it's start. 

To make sure that this is always true, we can restart the metrics-server pod once the sleep is done.

Reply:
> 

yeah, Let me try to test

Reply:
Why restart?

Reply:
> In fact, the cpu consumption is as follows
> 
> time elapse: 118.769µs cpu real consumption: 1436.7816091952816
> time elapse: 11.471613ms cpu real consumption: 29.770100399663576
> ...

Where this data comes from?

Reply:
> 
@serathius 
This is just to verify that the issue is not caused by metrics-server.
This is a suggestion made by @dgrisonnet .
```
To make sure that this is always true, we can restart the metrics-server pod once the sleep is done.
```
I will revert this part of the modification
But now after adding restart, other use cases fail, I'm still researching

Reply:
> > In fact, the cpu consumption is as follows
> > time elapse: 118.769µs cpu real consumption: 1436.7816091952816
> > time elapse: 11.471613ms cpu real consumption: 29.770100399663576
> > ...
> 
> Where this data comes from?

@serathius, This is a unit test I added myself  for the resource consumer code

Reply:
> Why restart?

To start from a fresh state. Isn't that what you wanted to make sure with your initial comment?
With a restart we can make sure that metrics-server starts fresh once the environment is correctly setup.

Reply:
> > Why restart?
> 
> To start from a fresh state. Isn't that what you wanted to make sure with your initial comment? With a restart we can make sure that metrics-server starts fresh once the environment is correctly setup.

Yes. I will continue to confirm whether the modification takes effect.
But now the test case `exposes prometheus metrics` keeps fail.  I don't know what caused
Isn't the metrics data updated in real time?